examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,351 @@
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId
+    ) {
+        $budgetResourceName = self::createCampaignBudget($googleAdsClient, $customerId);
+        $campaignResourceName = self::createCampaign(
+            $googleAdsClient,
+            $customerId,
+            $budgetResourceName
+        );
+        $adGroupResourceName = self::createAdGroup(
+            $googleAdsClient,
+            $customerId,
+            $campaignResourceName
+        );
+        self::createExpandedDSA($googleAdsClient, $customerId, $adGroupResourceName);
+        self::createWebPageCriteria($googleAdsClient, $customerId, $adGroupResourceName);
+    }
+
+    /**
+     * Creates a campaign budget.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @return string the campaign budget resource name
+     */
+    private static function createCampaignBudget(GoogleAdsClient $googleAdsClient, $customerId)
+    {
+        $campaignBudget = new CampaignBudget([
+            'name' => new StringValue(['value' => 'Interplanetary Cruise Budget #' . uniqid()]),
+            'delivery_method' => BudgetDeliveryMethod::STANDARD,
+            'amount_micros' => new Int64Value(['value' => 500000])
+        ]);
+
+        // Creates a campaign budget operation.
+        $campaignBudgetOperation = new CampaignBudgetOperation();
+        $campaignBudgetOperation->setCreate($campaignBudget);
+     
+        // Issues a mutate request to add campaign budgets.
+        $campaignBudgetServiceClient = $googleAdsClient->getCampaignBudgetServiceClient();
+        /** @var MutateCampaignBudgetsResponse $campaignBudgetResponse */
+        $campaignBudgetResponse = $campaignBudgetServiceClient->mutateCampaignBudgets(
+            $customerId,
+            [$campaignBudgetOperation]
+        );
+
+        $campaignBudgetResourceName = $campaignBudgetResponse->getResults()[0]->getResourceName();
+
+        printf("Added budget named '%s'.%s", $campaignBudgetResourceName, PHP_EOL);
+
+        return $campaignBudgetResourceName;
+    }
+
+    /**
+     * Creates a campaign.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param string $campaignBudgetResourceName the resource name of the campaign budget
+     * @return string the resource name of the newly created campaign
+     */
+    private static function createCampaign(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignBudgetResourceName
+    ) {
+        $startDate = new StringValue(['value' => date('Ymd', strtotime('+1 day'))]);
```

Comment:
nit: I would add a quick comment describing the range in words

Reply:
Done.